docs/news.rst

```diff
@@ -3,6 +3,155 @@
 Release notes
 =============
 
+Scrapy 1.4.0 (2017-XX-XX)
+-------------------------
+
+Scrapy 1.4 does not bring that many breathtaking new features
+but quite a few handy improvements nonetheless.
+
+Scrapy now supports anonymous FTP sessions with customizable user and
+password via the new :setting:`FTP_USER` and :setting:`FTP_PASSWORD` settings.
+**And if you're using Twisted version 17.1.0 or above, FTP is now available
+with Python 3.**
+
+Link extractors now work similarly to what a regular modern browser would
+do. Especially, leading and trailing whitespace are removed from attributes
+(think ``href="   http://example.com"``) when building ``Link`` objects.
+This whitespace-stripping also happens for ``action`` attributes with
+``FormRequest``.
+**Please also note that link extractors do not canonicalize URLs by default
+anymore.** This was puzzling users every now and then, and it's not what
+browsers do in fact, so we removed that extra transformation on extractred
+links.
+
+There's a new ``response.follow()`` shortcut for creating requests directly
+from a response instance and a relative URL.
+For example, instead of::
+
+    scrapy.Request(response.urljoin(somehrefvalue))
```

Comment:
Response.follow doesn't allow passing Selectors; Response.follow docs mention that TextResponse.follow alllows that, and https://doc.scrapy.org/en/master/topics/request-response.html#scrapy.http.TextResponse.follow mentions selectors and other stuff.

I think a verbose notice in the changelog is worth adding, as it is a huge change in recommended Scrapy API - starting with Scrapy 1.4, instead of scrapy.Request in most cases users should use response.follow, as it is more correct.